### DISCUSSION TITLE
Can't find eccodes.dll

### DISCUSSION THREAD

**Discussion by ghost** [blaylockbk/Herbie#380] [Q&A] [answered]
Conda/pip install not working on Windows 11? After running install, receiving error message: cannot load  library eccodes.dll. I added path to environment variables and tried numerous other options in different discussion forums ...

**Comment by williamhobbs** [comment, marked as answer]
_2025-08-05 Update/Edit: eccodes can be installed with pip now. See https://github.com/blaylockbk/Herbie/discussions/428. So the info below is now out of date..._ 

Did you install with `pip` or with `conda`? I can't get `pip` to work on Windows, but `conda` works fine. It looks like ECMWF only distributes Windows binaries for eccodes on conda-forge, so to use `pip` you either have to compile it yourself or find unofficial binaries. 

I use [miniforge](https://github.com/conda-forge/miniforge) and running the following in the Miniforge Prompt works for me:

```
conda create -n herbie_test_env python=3.12 -y
conda activate herbie_test_env
conda install -c conda-forge herbie-data
```

The regular Anaconda channel/repo might not work, so if you are using `conda` via Anaconda, you may have to specify the conda-forge channel (with `-c conda-forge` in the above commands). `conda` via Miniforge only uses the conda-forge channel, so I think it's optional to specify the channel in that case. I don't have Anaconda, so I'm just guessing on this...

**Reply by ghost** [reply]
Ty! I am using miniforge and that did the trick, appreciate the help!

**Comment by aniruddhaadak80** [comment]
Hi @jnapoliFEP ,


Hey there! Sorry to hear you're having trouble with `eccodes.dll` on Windows 11. Here are a few things you can try:

1. **Check Installation**: Make sure `eccodes` installed correctly. Run `conda list eccodes` or `pip show eccodes` to confirm.

2. **Update PATH**: Ensure the folder containing `eccodes.dll` is in your PATH. You can add it via System Properties > Environment Variables.

3. **Use Anaconda Prompt**: If you're using Anaconda, try running your script from the Anaconda Prompt instead of the regular Command Prompt.

4. **Reinstall**: Sometimes a reinstall helps. Try:
   ```bash
   conda uninstall eccodes
   conda install -c conda-forge eccodes
   ```

5. **Check Architecture**: Ensure your Python version matches the architecture (32-bit or 64-bit) of `eccodes.dll`.

This might helps . 

Thank you .

**Reply by ghost** [reply]
Ty, out of curiosity I will give this a try on my existing environment
